src/sql/src/pure/sql_server.rs

```diff
@@ -164,6 +165,23 @@ pub(super) async fn purify_source_exports(
         })
         .collect();
 
+    // If CDC is freshly enabled for a table, it has been observed that
+    // the `start_lsn`` from `cdc.change_tables` can be ahead of the LSN
+    // returned by `sys.fn_cdc_get_max_lsn`.  Eventually, the LSN returned
+    // by `sys.fn_cdc_get_max_lsn` will surpass `start_lsn`. For this
+    // reason, we choose the initial_lsn to be
+    // `max(start_lsns, sys.fn_cdc_get_max_lsn())``.
+    let mut possible_lsns = mz_sql_server_util::inspect::get_min_lsns(
+        client,
+        capture_instances.values().map(|instance| instance.as_ref()),
+    )
+    .await?;
+    possible_lsns.push(mz_sql_server_util::inspect::get_max_lsn_retry(client, timeout).await?);
+    let initial_lsn = possible_lsns
+        .iter()
+        .max()
+        .ok_or_else(SqlServerSourcePurificationError::NoInitialLsn)?;
```

Comment:
This is an unreachable error since the list `possible_lsns` contains at least one element from line 179. (if that fails execution doesn't reach this point).

But also, I don't think we want to be mixing the minimum LSNs between capture instances. If one capture instance's min LSN is way ahead of `max_lsn` for some reason we don't want to set the `initial_lsn` of every other table to that point. Do we gain something by forcing everyone to start at the same point?